[quick_actions_plaform_interface] add localizedSubtitle

sinyu1012 opened the pull request:
This is prequel PR for: https://github.com/flutter/packages/pull/8038
Containing only changes to quick_action_platform_interface package.

Add the localizedSubtitle field on quick actions for iOS

https://github.com/flutter/flutter/issues/129759

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

stuartmorgan commented on packages/quick_actions/quick_actions_platform_interface/CHANGELOG.md:
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 1.0.7
```
Adding new public API is a minor version under semver.

stuartmorgan reviewed (changes requested):
> [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version).

There is no such change in the current PR.

sinyu1012 commented:
> > [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version).
> 
> There is no such change in the current PR.

I've made the requested changes and updated the next version number in the changelog to 1.1.0. Please review when you have a chance.

stuartmorgan commented:
> > > [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning), or this PR is [exempt from version changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version).
> > 
> > 
> > There is no such change in the current PR.
> 
> I've made the requested changes

If you have made that change, you haven't uploaded it to your branch; there is no change to the version in the pubspec.yaml. Please see the `Files changed` tab, as well as the `Details` of the failure of `Linux repo_checks`.

sinyu1012 commented:
@stuartmorgan  Thank you for reminding me! I have completed the relevant modifications.

stuartmorgan reviewed (approved):
LGTM

bparrishMines reviewed (approved):
LGTM
